src/render/mod.rs

```diff
@@ -945,15 +945,18 @@ impl Renderer {
             let mx_world: mint::ColumnMatrix4<_> = Matrix4::from(w.world_transform).into();
             let pso_data = material.to_pso_data();
 
-            if let Some(ref key) = gpu_data.instance_cache_key {
-                let uv_range = [0.0; 4];
-                match pso_data {
-                    PsoData::Basic { color, param0, .. } => {
+            let instance = match pso_data {
+                PsoData::Basic { color, map, param0 } => {
+                    let uv_range = match map {
+                        Some(ref map) => map.uv_range(),
+                        None => [0.0; 4],
+                    };
+                    if let Some(ref key) = gpu_data.instance_cache_key {
                         let vec = self.instance_cache.entry(key.clone()).or_insert((
                             InstanceData {
                                 slice: gpu_data.slice.clone(),
                                 vertices: gpu_data.vertices.clone(),
-                                pso_data: pso_data.clone(),
+                                pso_data: PsoData::Basic { color, map, param0 },
```

Comment:
I can't understand this change, but okay.

Reply:
it's basically `pso_data` that has been destructured, so we compose it again
I'll add a comment